File: src/codes/Users_Manual/advanced/phase_filter.cpp
//~#include<autd3.hpp>
//~#include<autd3/link/nop.hpp>
//~int main() {
//~auto autd =
//~autd3::ControllerBuilder().add_device(autd3::AUTD3(autd3::Vector3::Zero())).open(autd3::link::Nop::builder());
autd.send(autd3::PhaseFilter::additive([](const auto& dev) {
  return [](const auto& tr) { return autd3::Phase(0); };
}));
//~return 0; }

File: src/codes/Users_Manual/advanced/phase_filter.rs
# extern crate autd3;
# extern crate tokio;
# use autd3::prelude::*;
# #[allow(unused_variables)]
# #[tokio::main]
# async fn main() -> Result<(), Box<dyn std::error::Error>> {
# let mut autd = Controller::builder().add_device(AUTD3::new(Vector3::zeros())).open(autd3::link::Nop::builder()).await?;
autd.send(PhaseFilter::additive(|_dev| |_tr| 0. * rad)).await?;
# Ok(())
# }